includes/actions.php

```diff
@@ -247,4 +248,30 @@ function give_set_donation_levels_max_min_amount( $form_id ) {
 	give_update_meta( $form_id, '_give_levels_maximum_amount', $max_amount? give_sanitize_amount_for_db( $max_amount ) : 0 );
 }
 
-add_action( 'give_pre_process_give_forms_meta', 'give_set_donation_levels_max_min_amount', 30 );
\ No newline at end of file
+add_action( 'give_pre_process_give_forms_meta', 'give_set_donation_levels_max_min_amount', 30 );
+
+
+/**
+ * Save donor address when donation complete
+ *
+ * @since 2.0
+ *
+ * @param int $payment_id
+ */
+function _give_save_donor_billing_address( $payment_id ) {
```

Comment:
@DevinWalker On payment save, we will save donor billing address to donor meta if unique